skrf/calibration/deembedding.py

```diff
@@ -908,4 +912,834 @@ def deembed(self, ntwk):
+    fixture-dut-fixture measurement.
+
+    This method is applicable only when there is a 2xthru measurement.
+
+    Example
+    --------
+    >>> import skrf as rf
+    >>> from skrf.calibration import Ieeep370_2xthru
+
+    Create network objects for dummy structure and dut
+
+    >>> s2xthru = rf.Network('2xthru.s2p')
+    >>> fdf = rf.Network('f-dut-f.s2p')
+
+    Create de-embedding object
+
+    >>> dm = Ieeep370_2xthru(dummy_2xthru = s2xthru, name = '2xthru')
+
+    Remove parasitics to get the actual device network
+
+    >>> dut = dm.deembed(fdf)
+    """
+    def __init__(self, dummy_2xthru, name=None, *args, **kwargs):
+        """
+        Ieeep370_2xthru De-embedding Initializer
+
+        Parameters
+        -----------
+
+        dummy_2xthru : :class:`~skrf.network.Network` object
+            Measurement of the 2x thru.
+
+        name : string
+            Optional name of de-embedding object
+
+        args, kwargs:
+            Passed to :func:`Deembedding.__init__`
+
+        See Also
+        ---------
+        :func:`Deembedding.__init__`
+
+        """
+        self.s2xthru = dummy_2xthru.copy()
+        dummies = [self.s2xthru]
+
+        Deembedding.__init__(self, dummies, name, *args, **kwargs)
+        self.s_side1, self.s_side2 = self.split2xthru(self.s2xthru)
+
+    def deembed(self, ntwk):
+        """
+        Perform the de-embedding calculation
+
+        Parameters
+        ----------
+        ntwk : :class:`~skrf.network.Network` object
+            Network data of device measurement from which
+            thru fixtures needs to be removed via de-embedding
+
+        Returns
+        -------
+        caled : :class:`~skrf.network.Network` object
+            Network data of the device after de-embedding
+
+        """
+
+        # check if the frequencies match with dummy frequencies
+        if ntwk.frequency != self.s2xthru.frequency:
+            raise(ValueError('Network frequencies dont match dummy frequencies.'))
+
+        # TODO: attempt to interpolate if frequencies do not match
+
+        return self.s_side1.inv ** ntwk ** self.s_side2.inv
+    
+    def dc_interp(self, s, f):
+        """
+        enforces symmetric upon the first 10 points and interpolates the DC
+        point.
+        """ 
+        sp = s[0:9]
+        fp = f[0:9]
+        
+        snp = concatenate((conj(flip(sp)), sp))
+        fnp = concatenate((-1*flip(fp), fp))
+        # mhuser : used cubic instead spline (not implemented)
+        snew = interp1d(fnp, snp, axis=0, kind = 'cubic')
+        return real(snew(0))
+    
+    def COM_receiver_noise_filter(self, f,fr):
+        """
+        receiver filter in COM defined by eq 93A-20
+        """ 
+        fdfr = f / fr
+        # eq 93A-20
+        return 1 / (1 - 3.414214 * fdfr**2 + fdfr**4 + 1j*2.613126*(fdfr - fdfr**3))
+    
+    
+    def makeStep(self, impulse):
+        """
+        mhuser : no need to call step function here, cumsum will be enough and efficient

```

Comment:
I'd rather put this comment into a comment in a code rather than in the docstring.

Reply:
Commented